src/lib/network_pool/transaction_pool.ml

```diff
@@ -1106,20 +1083,7 @@ struct
           , Intf.Verification_error.t )
           Deferred.Result.t =
         let open Deferred.Result.Let_syntax in
-        let is_sender_local =
-          Envelope.Sender.(equal Local) (Envelope.Incoming.sender diff)
-        in
         let open Intf.Verification_error in
-        let%bind () =
-          (* TODO: we should probably remove this -- the libp2p gossip cache should cover this already (#11704) *)
-          let (`Already_mem already_mem) =
-            Lru_cache.add t.recently_seen (Lru_cache.T.hash diff.data)
-          in
-          Deferred.return
-          @@ Result.ok_if_true
-               ((not already_mem) || is_sender_local)
-               ~error:Recently_seen
```

Comment:
If we don't also use it elsewhere, it would be good to remove this error too (unless this error propagates over the protocol boundary to other nodes). Can be another PR though.

Reply:
removed it!